### PR TITLE
Smack JingleReason#toXML must include both text and element

### PR DESCRIPTION
**Pull request by cmeng-git**
The JingleReason#toXML method must include both the optional text string and element XmlElement in the create process.

### CODE REVIEW

**Comment by cmeng-git** [comment]
I have already change <tt>null</tt> to <code>null</code> as per Smack check requirements. But still get rejected. Anything else I need to modifiy.

**Comment by Flowdalic** [comment]
Thanks, from a quick look, there are at least two things you may want to change: First, this should target the 4.4 branch. Secondly, not all HTML tags are supported by javadoc. I the case where you used `tt`, you may want to use `{@link JingleReason}` instead.

File: smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/element/JingleReason.java [hunk 1]
```diff
@@ -155,6 +179,14 @@ public XmlStringBuilder toXML(XmlEnvironment enclosingXmlEnvironment) {
 
         xml.emptyElement(reason);
 
+        // add reason "text" if we have it
+        xml.optElement(TEXT_ELEMENT, text);
+
+        // add the extra element if it has been specified.
+        if (element != null) {
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/jingle/element/JingleReason.java:186]: This should be `xml.optAppend(element)` instead.

**Comment by cmeng-git** [comment]
Closed pull request. Will try to do it in 4.4 branch

**Comment by Flowdalic** [comment]
Please note that you don't have to close PRs to update the target branch.

**Comment by cmeng-git** [comment]
Thank for the info. I am still learning all the github features.

**Pull request closed without merging**